sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerAccessor.java

```diff
@@ -109,20 +112,26 @@ public static SpannerAccessor getOrCreate(SpannerConfig spannerConfig) {
   private static SpannerAccessor createAndConnect(SpannerConfig spannerConfig) {
     SpannerOptions.Builder builder = SpannerOptions.newBuilder();
 
-    // Set retryable codes for all API methods
+    Set<Code> retryableCodes = new HashSet<>();
```

Comment:
You can move this into the else block

Reply:
spannerConfig.getRetryableCodes() returns an ImmutableSet. So, I have to create a new HashSet and then add all the retryable codes for these 2 conditions.